arch_build.cmake

```diff
@@ -0,0 +1,28 @@
+###########################################################
+#
+# HS App platform build setup
+#
+# This file is evaluated as part of the "prepare" stage
+# and can be used to set up prerequisites for the build,
+# such as generating header files
+#
+###########################################################
+
+# The list of header files that control the HS configuration
+set(HS_PLATFORM_CONFIG_FILE_LIST
+  hs_internal_cfg.h
+  hs_custom.h
+  hs_msgids.h
+  hs_platform_cfg.h
+)
+
+# Create wrappers around the all the config header files
+# This makes them individually overridable by the missions, without modifying
+# the distribution default copies
+foreach(HS_CFGFILE ${HS_PLATFORM_CONFIG_FILE_LIST})
+  set(DEFAULT_SOURCE "${CMAKE_CURRENT_LIST_DIR}/config/default_${HS_CFGFILE}")
```

Comment:
Consider consolidating the target_defs search patterns.
Tables has: https://github.com/nasa/cFE/blob/7af467efdc15f788d44d67cb807403b10c4f0c99/cmake/arch_build.cmake#L187-L201.
Then there's the file list:
https://github.com/nasa/cFE/blob/7af467efdc15f788d44d67cb807403b10c4f0c99/cmake/arch_build.cmake#L703-L712
Maybe there's more?  Would be nice if the search logic was intuitive, consistent, and well documented.